### PR TITLE
Tests for Shortcuts

### PR DESCRIPTION
**Pull request by FillBk**
Tests for shortcuts: 

- [x] F
- [x] Shift+F
- [x] G
- [ ] H
- [x] W
- [x] S
- [x] Shift+S
- [x] I
- [x] O
- [x] X
- [x] . 
- [x] 2
- [ ] Ins
- [x] Ctrl+S
- [x] Ctrl+Shift+S
- [x] Shift+**⇐**
- [x] Shift+**⇒**
- [x] Shift+**⇑**
- [x] Shift+**⇓**
- [x] C
- [x] **⇐**
- [x] **⇒**
- [x] R
- [x] Y
- [x] Ctrl+I
- [x] Alt+I
- [x] Ctrl+H
- [x] Ctrl+B
- [x] Ctrl+P
- [x] Alt+R
- [x] Alt+S
- [x] Shift+C
- [x] Shift+T
- [x] Shift+V
- [x] Shift+H
- [x] Shift+Alt+F
- [x] Shift+L
- [x] M

...

### CODE REVIEW

File: test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py [hunk 1]
```diff
@@ -231,3 +236,264 @@ def test_Plot_Plot_Shortcut_Key_R(self):
...
         clear_pixmap = plot.plot.viewport().grab()
         self.assertTrue(Pixmap.is_black(clear_pixmap))
+
+    def test_Plot_Plot_Shortcut_Key_F(self):
+        """
+        Test Scope:
+            Check if all plotted signals is fitted on Y ax after pressing key "F".
+        Events:
+            - Open 'FileWidget' with valid measurement.
+            - Select 2 measurement and create a plot
```
Comment (tov101) [inline test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py:246]: here you are actually talking about 2 channels? or something else?
2 measurement seems not so clear.

File: test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py [hunk 2]
```diff
@@ -231,3 +236,264 @@ def test_Plot_Plot_Shortcut_Key_R(self):
...
+            - Press Key "S"
```
Comment (tov101) [inline test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py:247]: Maybe one extra detail needs to be written here.
The precondition is that channels should not be streched on entire plot heigth.
'S' - 'Stack' command is used to ensure that channels are resized to fulfill the precondition.

File: test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py [hunk 3]
```diff
@@ -231,3 +236,264 @@ def test_Plot_Plot_Shortcut_Key_R(self):
...
+            QtTest.QTest.keyClick(plot.plot.viewport(), QtCore.Qt.Key_G)
+            self.processEvents()
+            self.assertTrue(plot.plot.x_axis.grid)
+            self.assertTrue(plot.plot.y_axis.grid)
+            # press key "G"
+            QtTest.QTest.keyClick(plot.plot.viewport(), QtCore.Qt.Key_G)
+            self.processEvents()
+            self.assertFalse(plot.plot.x_axis.grid)
+            self.assertFalse(plot.plot.y_axis.grid)
+        elif plot.plot.x_axis.grid and  not plot.plot.y_axis.grid:
```
Comment (tov101) [inline test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py:393]: What about creating 'subTest' here for each case?

File: test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py [hunk 4]
```diff
@@ -231,3 +236,264 @@ def test_Plot_Plot_Shortcut_Key_R(self):
...
+        """
+        # Event
+        self.setUpFileWidget(measurement_file=self.measurement_file, default=True)
+        # Press PushButton "Create Window"
+        self.create_window(window_type="Plot")
+        self.assertEqual(len(self.widget.mdi_area.subWindowList()), 1)
+        plot = self.widget.mdi_area.subWindowList()[0].widget()
+        channel_0 = self.add_channel_to_plot(plot=plot, channel_index=7)
+        channel_1 = self.add_channel_to_plot(plot=plot, channel_name="ASAM.M.SCALAR.UBYTE.HYPERBOLIC")
+        self.assertEqual(2, plot.channel_selection.topLevelItemCount())
```
Comment (tov101) [inline test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py:450]: This block seems to be reused often. Maybe you can include it in 'setUp' if it's common for all test cases.

File: test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py [hunk 5]
```diff
@@ -225,9 +223,253 @@ def test_Plot_Plot_Shortcut_Key_R(self):
...
+            - Press Key "S" to separate signals (precondition)
+            - Press Key "F"
+        Evaluate:
+            - Evaluate that two signals are available
+            - Evaluate that plot is not black
+            - Evaluate that the color of signals is not displayed on top and bottom line of plot after pressing key "S"
+            - Evaluate that are both colors of signals in top and bottom line of plot after pressing key "F"
+        """
+        channel_0 = self.add_channel_to_plot(plot=self.plot, channel_index=7)
+        channel_1 = self.add_channel_to_plot(plot=self.plot, channel_name="ASAM.M.SCALAR.UBYTE.HYPERBOLIC")
```
Comment (tov101) [inline test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py:249]: Do you still need these channels? It's possible to use the ones defined in `setUp` ?

Comment (FillBk) [reply test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py:249]: Nope, is one test without signals

Comment (FillBk) [reply test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py:249]: Key "R"

File: test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py [hunk 6]
```diff
@@ -225,9 +243,625 @@ def test_Plot_Plot_Shortcut_Key_R(self):
...
+            self.assertTrue(
+                colorMap[y].count(self.channel_36.color.name()),
+                f"Line {y} doesn't contain {self.channel_36.name} color",
+            )
+            if y < self.plot.plot.height() / 2:
+                self.assertFalse(
+                    colorMap[y].count(self.channel_37.color.name()),
+                    f"Line {y} doesn't contain {self.channel_37.name} color",
+                )
+
```
Comment (tov101) [inline test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py:366]: Same comments as above

File: test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py [hunk 7]
```diff
@@ -225,9 +243,625 @@ def test_Plot_Plot_Shortcut_Key_R(self):
...
+        # Create a color map for plot
+        colorMap = Pixmap.color_map(self.plot.plot.viewport().grab())
+        self.assertTrue(colorMap, "Failed to create color map")
+        # Evaluate if signal was separated
+        for y in range(self.plot.plot.height()):
+            if y < self.plot.plot.height() / 2:
+                self.assertFalse(
+                    colorMap[y].count(self.channel_37.color.name()),
+                    f"Line {y} contain {self.channel_37.name} color",
+                )
```
[the remark this answers is not in the document]
Comment (FillBk) [reply test/asammdf/gui/widgets/test_PlotWidget_Shortcuts.py]: This is not the scope of test

**Pull request closed without merging**